### ISSUE TITLE
[K8S] Provision on Kubernetes failed but `sky check` shows it is enabled

### ISSUE THREAD

**Issue by cblmemo** [skypilot-org/skypilot#3534]
On commit `7f30ce5e`, `sky check` shows k8s is enabled, but it failed later when i tried to provision a k8s cluster. Should we check the dependency of `netcat` in `sky check` as well?

```bash
$ sky check   
Checking credentials to enable clouds for SkyPilot.
  AWS: enabled                              
  Azure: enabled                              
  Cloudflare, for R2 object store: disabled                              
    Reason: [r2] profile is not set in ~/.cloudflare/r2.credentials. Additionally, Account ID from R2 dashboard is not set. Run the following commands:
      $ pip install boto3
      $ AWS_SHARED_CREDENTIALS_FILE=~/.cloudflare/r2.credentials aws configure --profile r2
      $ mkdir -p ~/.cloudflare
      $ echo <YOUR_ACCOUNT_ID_HERE> > ~/.cloudflare/accountid
    For more info: https://skypilot.readthedocs.io/en/latest/getting-started/installation.html#cloudflare-r2
  Cudo: disabled                              
    Reason: Cudo tools are not installed. Run the following commands:
      $ pip install cudo-compute
    [ModuleNotFoundError] No module named 'cudo_compute'
  Fluidstack: disabled                              
    Reason: Failed to access FluidStack Cloud with credentials. To configure credentials, go to:
      https://console.fluidstack.io 
    to obtain an API key and API Token, then add save the contents to ~/.fluidstack/api_key and ~/.fluidstack/api_token 

  Checking GCP...INFO:googleapiclient.discovery_cache:file_cache is only supported with oauth2client<4.0.0
  GCP: enabled                              
  IBM: disabled                              
    Reason: Missing credential file at /home/memory/.ibm/credentials.yaml.
    Store your API key and Resource Group id in ~/.ibm/credentials.yaml in the following format:
      iam_api_key: <IAM_API_KEY>
      resource_group_id: <RESOURCE_GROUP_ID>
  Kubernetes: enabled                              
    Hint: Could not detect GPU resources (`nvidia.com/gpu`) in Kubernetes cluster. If this cluster contains GPUs, please ensure GPU drivers are installed on the node. Check if the GPUs are setup correctly by running `kubectl describe nodes` and looking for the nvidia.com/gpu resource. Please refer to the documentation on how to set up GPUs.
  Lambda: enabled                              
  OCI: disabled                              
    Reason: Missing credential file at ~/.oci/config. For more details, refer to: https://skypilot.readthedocs.io/en/latest/getting-started/installation.html#oracle-cloud-infrastructure-oci
  Paperspace: disabled                              
    Reason: Failed to access Paperspace Cloud with credentials.
    To configure credentials, follow the instructions at: https://skypilot.readthedocs.io/en/latest/getting-started/installation.html#paperspace
    Generate API key and create a json at `~/.paperspace/config.json` with 
         {"apiKey": "[YOUR API KEY]"}
    Reason: Credentials not found
  RunPod: enabled                              
  SCP: disabled                              
    Reason: Failed to access SCP with credentials. To configure credentials, see: https://cloud.samsungsds.com/openapiguide
    Generate API key and add the following line to ~/.scp/scp_credential:
      access_key = [YOUR API ACCESS KEY]
      secret_key = [YOUR API SECRET KEY]
      project_id = [YOUR PROJECT ID]
  vSphere: disabled                              
    Reason: Missing credential file at /home/memory/.vsphere/credential.yaml.

To enable a cloud, follow the hints above and rerun: sky check
If any problems remain, refer to detailed docs at: https://skypilot.readthedocs.io/en/latest/getting-started/installation.html

🎉 Enabled clouds 🎉
  ✔ AWS
  ✔ Azure
  ✔ GCP
  ✔ Kubernetes
  ✔ Lambda
  ✔ RunPod
```

```bash
$ sky jobs launch --cloud kubernetes --cpus 1 -- echo hi
Task from command: echo hi
Managed job 'sky-cmd' will be launched on (estimated):
I 05-09 20:04:00 optimizer.py:694] == Optimizer ==
I 05-09 20:04:00 optimizer.py:717] Estimated cost: $0.0 / hour
I 05-09 20:04:00 optimizer.py:717] 
I 05-09 20:04:00 optimizer.py:842] Considered resources (1 node):
I 05-09 20:04:00 optimizer.py:912] ---------------------------------------------------------------------------------------------
I 05-09 20:04:00 optimizer.py:912]  CLOUD        INSTANCE    vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE   COST ($)   CHOSEN   
I 05-09 20:04:00 optimizer.py:912] ---------------------------------------------------------------------------------------------
I 05-09 20:04:00 optimizer.py:912]  Kubernetes   1CPU--1GB   1       1         -              kubernetes    0.00          ✔     
I 05-09 20:04:00 optimizer.py:912] ---------------------------------------------------------------------------------------------
I 05-09 20:04:00 optimizer.py:912] 
Launching a managed job 'sky-cmd'. Proceed? [Y/n]: 
Launching managed job 'sky-cmd' from jobs controller...
Launching jobs controller...
I 05-09 20:04:01 optimizer.py:694] == Optimizer ==
I 05-09 20:04:01 optimizer.py:717] Estimated cost: $0.0 / hour
I 05-09 20:04:01 optimizer.py:717] 
I 05-09 20:04:01 optimizer.py:842] Considered resources (1 node):
I 05-09 20:04:01 optimizer.py:912] ---------------------------------------------------------------------------------------------
I 05-09 20:04:01 optimizer.py:912]  CLOUD        INSTANCE    vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE   COST ($)   CHOSEN   
I 05-09 20:04:01 optimizer.py:912] ---------------------------------------------------------------------------------------------
I 05-09 20:04:01 optimizer.py:912]  Kubernetes   1CPU--3GB   1       3         -              kubernetes    0.00          ✔     
I 05-09 20:04:01 optimizer.py:912] ---------------------------------------------------------------------------------------------
I 05-09 20:04:01 optimizer.py:912] 
I 05-09 20:04:01 cloud_vm_ray_backend.py:4378] Creating a new cluster: 'sky-jobs-controller-402b1bba' [1x Kubernetes(1CPU--3GB, cpus=1, mem=3x, disk_size=50)].
I 05-09 20:04:01 cloud_vm_ray_backend.py:4378] Tip: to reuse an existing cluster, specify --cluster (-c). Run `sky status` to see existing clusters.
I 05-09 20:04:03 cloud_vm_ray_backend.py:1372] To view detailed progress: tail -n100 -f /home/memory/sky_logs/sky-2024-05-09-20-04-01-654473/provision.log

RuntimeError: `nc` is required to setup Kubernetes cloud with `portforward` default networking mode and it is not installed. On Debian/Ubuntu, install it with:
  $ sudo apt install netcat
On MacOS, install it with: 
  $ brew install netcat
```

_Version & Commit info:_
* `sky -v`: PLEASE_FILL_IN
* `sky -c`: PLEASE_FILL_IN

**Comment by romilbhardwaj** [comment]
Related to https://github.com/skypilot-org/skypilot/issues/3252, Will be fixed by https://github.com/skypilot-org/skypilot/pull/3306 :)

**Comment by cblmemo** [comment]
> Related to #3252, will be fixed by #3306 :)

Thanks for pointing it out! Closed due to duplication.
